### PR TITLE
Added  approx_max_k function to math.py with test function is also added

### PR DESCRIPTION
**Pull request by Shubham7-1**
# PR Description
Created `approx_max_k` function to math.py with test function is also added

## Related Issue
https://github.com/unifyai/ivy/issues/25774 

Close #

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [ ] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by Shubham7-1** [comment]
@umairjavaid  Could you please Review My PR

**Review by umairjavaid — CHANGES_REQUESTED** [review]
Hi @Shubham7-1, The tests are failing. You have to make sure all the tests pass

**Comment by umairjavaid** [comment]
Hi @Shubham7-1, The tests are failing. You have to make sure all the tests pass

```
    | Traceback (most recent call last):
    |   File "/ivy/ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py", line 3256, in test_tensorflow_approx_max_k
    |     helpers.test_frontend_function(
    |   File "/ivy/ivy_tests/test_ivy/helpers/function_testing.py", line 981, in test_frontend_function
    |     value_test(
    |   File "/ivy/ivy_tests/test_ivy/helpers/assertions.py", line 135, in value_test
    |     assert_same_type_and_shape([ret_np_flat, ret_np_from_gt_flat])
    |   File "/ivy/ivy_tests/test_ivy/helpers/assertions.py", line 74, in assert_same_type_and_shape
    |     assert (
    | AssertionError: returned dtype = int64, ground-truth returned dtype = int32
    | Falsifying example: test_tensorflow_approx_max_k(
    |     frontend='tensorflow',
    |     backend_fw='numpy',
    |     on_device='cpu',
    |     dtype_and_x=(['float32'],
    |      [array([-4.2813177e+00,  1.5000000e+00,  9.9998999e-01,  1.5000000e+00,
    |              -1.3524009e-02, -3.7381943e+01, -5.0000000e-01], dtype=float32)]),
    |     fn_tree='ivy.functional.frontends.tensorflow.math.approx_max_k',
    |     test_flags=FrontendFunctionTestFlags(
    |         num_positional_args=2,
    |         with_out=False,
    |         inplace=False,
    |         as_variable=[False],
    |         native_arrays=[False],
    |         test_compile=False,
    |         generate_frontend_arrays=True,
    |         transpile=False,
    |         precision_mode=False,
    |     ),
    | )
    | 
    | You can reproduce this example by temporarily adding @reproduce_failure('6.86.2', b'AXic42Zk4+VhYwACRobe3vMll+MEGRsUi+pP8ikIMTIyMTI0MIABIyMbMyPDAe6+p3q89f9RZBh4mBgZD2QAmZZP7jKwMjBGVoet2dtyoIexIag/bu+Ox20MrCxAFQxQwATUAgBy1xiD') as a decorator on your test case
    
```

**Comment by Shubham7-1** [comment]
Hii @umairjavaid  I had Done changes to test function Could you please Review it

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 1]
```diff
@@ -3230,3 +3230,43 @@ def test_tensorflow_zeta(
...
+    *,
+    dtype_and_x,
+    frontend,
+    test_flags,
+    fn_tree,
+    backend_fw,
+    on_device,
+):
+    input_dtype, x = dtype_and_x
+    k = 5
```
Comment (umairjavaid) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:3255]: Why have you hard coded the value of `k` in the test function?

Comment (umairjavaid) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:3255]: You need to define an `int` strategy for 'k'

**Review by umairjavaid — CHANGES_REQUESTED** [review]

**Comment by Shubham7-1** [comment]
Hii @umairjavaid I had make changes according to feedback
Is it now good fit ?

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 2]
```diff
@@ -3230,3 +3230,46 @@ def test_tensorflow_zeta(
         x=x[0],
         q=x[1],
     )
+ #approx_max_k
+@handle_frontend_test(
+    fn_tree="tensorflow.math.approx_max_k",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        large_abs_safety_factor=24,
```
Comment (umairjavaid) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:3238]: Why have you hard coded `large_abs_safety_factor`

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 3]
```diff
@@ -3230,3 +3230,46 @@ def test_tensorflow_zeta(
...
+        small_abs_safety_factor=24,
```
Comment (umairjavaid) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:3239]: Why have you hard coded `small_abs_safety_factor`

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 4]
```diff
@@ -3230,3 +3230,46 @@ def test_tensorflow_zeta(
...
+        safety_factor_scale="log",
+        min_num_dims=1,
```
Comment (umairjavaid) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:3241]: Why have you hard coded `min_num_dims`

**Review by umairjavaid — CHANGES_REQUESTED** [review]

**Pull request closed without merging**